ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorUtil.java

```diff
@@ -565,4 +566,16 @@ public static Map<String, Integer> getPoolConf(HiveConf hiveConf) {
     }
     return poolConf;
   }
+
+  public static String getCompactPoolName(HiveConf conf, org.apache.hadoop.hive.ql.metadata.Table table) 
```

Comment:
extract the `getPoolName` method from Initiator to some common place. Note `resolveDatabase` is not cheap, so it has to be cached.

Reply:
Done